### PR TITLE
Add a readme.md to the models dir. Rename the transformers corpora.

### LINKED ISSUE #24: Models readme
Add a descriptive readme.md file to `models` explaining the organization of the corpora, structure of a model file, etc.

### PR DESCRIPTION
**Pull request by jeremyfowers**
Closes #24 

Don't be alarmed by the large list of changes! I just renamed two corpora to fit the naming scheme in the documentation.

The only file that needs to be reviewed in detail is models/readme.md.

### CODE REVIEW

File: models/readme.md [hunk 1]
```diff
@@ -1 +1,164 @@
...
+- `graph_convolutions`: Graph Neural Network (GNN) models from a variety of publications. See the docstring on each .py file for the source.
+- `popular_on_huggingface`: hundreds of the most-downloaded models from the [Huggingface models repository](https://huggingface.co/models).
+- `selftest`: a small corpus with small models that can be used for testing out the MLAgility tools.
+- `torch_hub`: a variety of models, including many image classification models, from the [Torch Hub repository](https://github.com/pytorch/hub).
+- `torchvision`: image recognition models from the [`torchvision` library](https://pytorch.org/vision/stable/index.html).
+  - _Note_: the `torchvision` library also includes many image classification models, but we excluded them to avoid overlap between our `torch_hub` and `torchvision` corpora.
+- `transformers_keras`: Transformer models from the [Huggingface `transformers` library](https://huggingface.co/docs/transformers/index), expressed in the Keras framework.
+- `transformers_pytorch`: Transformer models from the [Huggingface `transformers` library](https://huggingface.co/docs/transformers/index), expressed in the PyTorch framework.
```
Comment (danielholanda) [inline models/readme.md:24]: There is no one-to-one match between the `transformers_keras` and the `transformers_pytorch` corpora. However, overlaps do exist.

File: models/readme.md [hunk 2]
```diff
@@ -1 +1,164 @@
...
+
+## Table of Contents
+
+- [Benchmark Organization](#benchmark-organization)
+- [Running the Benchmark](#running-the-benchmark)
+- [Model Template](#model-template)
+
+## Benchmark Organization
+
+The MLAgility benchmark is made up of several corpora of models (_corpora_ is the plural of _corpus_... we had to look it up too). Each corpus is named after the online repository that the models were sourced from. Each corpus gets its own subdirectory in the `models` directory.
```
Comment (danielholanda) [inline models/readme.md:13]: Do we want to create a distinction between scripts and models (each script may instantiate multiple models)?

File: models/readme.md [hunk 3]
```diff
@@ -1 +1,164 @@
...
+You can aggregate all of the benchmarking results from your `mlagility cache` into a CSV file with:
+
+```
+benchit report
+```
+
+If you want to only report on a subset of models, we recommend benchmarking those models into a specific cache directory:
+
+```
+# Benchmark the models into a specific cache directory
```
Comment (danielholanda) [inline models/readme.md:60]:
```suggestion
# Save benchmarked models into a specific cache directory
```

File: models/readme.md [hunk 4]
```diff
@@ -1 +1,164 @@
...
+benchit benchmark --all --search-dir selftest
+```
+
+You can aggregate all of the benchmarking results from your `mlagility cache` into a CSV file with:
+
+```
+benchit report
+```
+
+If you want to only report on a subset of models, we recommend benchmarking those models into a specific cache directory:
```
Comment (danielholanda) [inline models/readme.md:57]:
```suggestion
If you want to only report on a subset of models, we recommend saving the benchmarking results into a specific cache directory:
```
Suggestion for clarity

File: models/readme.md [hunk 5]
```diff
@@ -1 +1,164 @@
...
+MLAgility supports parameterizing models so that you can sweep over interesting properties such as batch size, sequence length, image size, etc. The `mlagility.parser.parse()` method parses a set of standardized parameters that we have defined and provides them to the model as it is instantiated.
+
+For example, this code would retrieve user-defined `batch_size` and `max_seq_length` (maximum sequence length) parameters.
+
+```
+from mlagility.parser import parse
+parse(["batch_size", "max_seq_length"])
+```
+
+You can pass parameters into a benchmarking run with the `--script-args` argument to `benchit`. For example, the command:
```
Comment (danielholanda) [inline models/readme.md:106]: Might be useful to explain that those CLI arguments may also be used outside of benchit

Example `bert.py --batch-size 8`

File: models/readme.md [hunk 6]
```diff
@@ -1 +1,164 @@
...
+
+```
+benchit bert.py --search-dir transformers_pytorch --script-args="--batch-size=8,--max-seq-length=128"
```
Comment (danielholanda) [inline models/readme.md:109]:
```suggestion
benchit bert.py --search-dir transformers_pytorch --script-args="--batch-size 8 --max-seq-length 128"
```

File: models/readme.md [hunk 7]
```diff
@@ -1 +1,164 @@
...
+benchit report --cache-dir selftest_results
+```
+
+## Model Template
+
+Each model in the MLAgility benchmark follows a template to keep things consistent. This template is meant to be as simple as possible. The models themselves should also be instantiated and called in a completely vendor-neutral manner.
+
+### The model.py File
+
+Each model in the MLAgility benchmark is hosted in a `model.py` file. This file is a Python script that must meet the following requirements:
```
Comment (danielholanda) [inline models/readme.md:73]: Is the idea to only allow scripts in mlagility if that script only instantiates a single model?

File: models/readme.md [hunk 8]
```diff
@@ -1 +1,164 @@
...
+    - "max_audio_seq_length": Arg("max_audio_seq_length", default=25600, type=int),
+    - "height": Arg("height", default=224, type=int),
+    - "num_channels": Arg("num_channels", default=3, type=int),
+    - "width": Arg("width", default=224, type=int),
+- Args for Graph Neural Networks
+    - "k": Arg("k", default=8, type=int),
+    - "alpha": Arg("alpha", default=2.2, type=float),
+    - "out_channels": Arg("out_channels", default=16, type=int),
+    - "num_layers": Arg("num_layers", default=8, type=int),
+    - "in_channels": Arg("in_channels", default=1433, type=int),-
```
Comment (danielholanda) [inline models/readme.md:129]: We might want to review those eventually, as a few of those GNN parameters are only used in 1-2 models and are not very descriptive.

**Review by danielholanda — CHANGES_REQUESTED** [review]

File: models/readme.md (hunk 1 above)
Comment (jeremyfowers) [reply models/readme.md:24]: Added!

File: models/readme.md (hunk 2 above)
Comment (jeremyfowers) [reply models/readme.md:13]: Yes, added!

File: models/readme.md (hunk 3 above)
Comment (jeremyfowers) [reply models/readme.md:60]: Made an edit like this

File: models/readme.md (hunk 4 above)
Comment (jeremyfowers) [reply models/readme.md:57]: nice

File: models/readme.md (hunk 5 above)
Comment (jeremyfowers) [reply models/readme.md:106]: Added!

File: models/readme.md (hunk 6 above)
Comment (jeremyfowers) [reply models/readme.md:109]: good catch thanks

File: models/readme.md (hunk 7 above)
Comment (jeremyfowers) [reply models/readme.md:73]: Nope, reworked the definition

File: models/readme.md (hunk 8 above)
Comment (jeremyfowers) [reply models/readme.md:129]: Yeah they should probably have comments or something

Comment (jeremyfowers) [reply models/readme.md:129]: I added definitions for the general args. Made an issue for @ramkrishna2910 to add definitions to the GNN args https://github.com/groq/mlagility/issues/78

**Review by danielholanda — APPROVED** [review]

**Pull request merged**